src/plugins/proxy/src/plugin.cpp

```diff
@@ -0,0 +1,625 @@
+    return std::dynamic_pointer_cast<ov::ICompiledModel>(compiled_model);
+}
+
+std::shared_ptr<ov::ICompiledModel> ov::proxy::Plugin::compile_model(const std::shared_ptr<const ov::Model>& model,
+                                                                     const ov::AnyMap& properties,
+                                                                     const ov::RemoteContext& context) const {
+    auto ctx = ov::proxy::RemoteContext::get_hardware_context(context);
+    auto dev_name = ctx.get_device_name();
+    auto device_config = construct_device_config(dev_name, m_configs, properties);
+
+    std::shared_ptr<const ov::IPlugin> plugin = shared_from_this();
+    auto compiled_model =
+        std::make_shared<ov::proxy::CompiledModel>(get_core()->compile_model(model, ctx, device_config),
+                                                   plugin,
+                                                   context);
+    return std::dynamic_pointer_cast<ov::ICompiledModel>(compiled_model);
+}
+
+std::shared_ptr<ov::IRemoteContext> ov::proxy::Plugin::create_context(const ov::AnyMap& remote_properties) const {
+    // TODO: if no device id, try to create context for each plugin
+    auto dev_name = get_device_name();
+    auto dev_idx = get_device_from_config(remote_properties);
+    auto has_dev_idx = is_device_in_config(remote_properties);
+    auto is_new_api = get_core()->is_new_api();
+
+    auto device_config = remote_properties;
+    remove_proxy_properties(device_config);
+
+    if (has_dev_idx) {
+        auto remote_context = std::make_shared<ov::proxy::RemoteContext>(
+            get_core()->create_context(get_fallback_device(get_device_from_config(remote_properties)), device_config),
+            dev_name,
+            dev_idx,
+            has_dev_idx,
+            is_new_api);
+        return std::dynamic_pointer_cast<ov::IRemoteContext>(remote_context);
+    }
+    // Properties doesn't have device id, so try to create context for all devices
+    const auto hidden_devices = get_hidden_devices();
+    for (size_t i = 0; i < hidden_devices.size(); i++) {
+        try {
+            auto remote_context = std::make_shared<ov::proxy::RemoteContext>(
+                get_core()->create_context(get_fallback_device(get_device_from_config(remote_properties)),
+                                           device_config),
+                dev_name,
+                i,
+                has_dev_idx,
+                is_new_api);
+            return std::dynamic_pointer_cast<ov::IRemoteContext>(remote_context);
+        } catch (const ov::Exception&) {
+        }
+    }
+    std::string properties_str = "{";
+    bool has_elements = false;
+    for (const auto& it : remote_properties) {
+        if (has_elements)
+            properties_str += ", ";
+        else
+            has_elements = true;
+        properties_str += it.first + ": " + it.second.as<std::string>();
+    }
+    properties_str += "}";
```

Comment:
why do we need custom initialization? 
We can write `ov::Any(remote_properties).as<std::string>()`